### PR TITLE
Update pooling strategy for embeding models

### PR DESCRIPTION
**Pull request by ravi03071991**
# Description

PR to update pooling strategy for embedding models based on config file from HF.

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [ ] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama_index/embeddings/huggingface_utils.py [hunk 1]
```diff
@@ -72,3 +74,26 @@ def format_text(
...
+    pooling_config_url = (
+        f"https://huggingface.co/{model_name}/raw/main/1_Pooling/config.json"
+    )
+
+    try:
+        response = requests.get(pooling_config_url)
+        config_data = response.json()
+
+        cls_token = config_data.get("pooling_mode_cls_token", False)
+        mean_tokens = config_data.get("pooling_mode_mean_tokens", False)
```
Comment (hatianzhang) [inline llama_index/embeddings/huggingface_utils.py:89]: so every model card will have those 2 flag with the same name always?

Comment (ravi03071991) [reply llama_index/embeddings/huggingface_utils.py:89]: pooling_mode_cls_token, pooling_mode_mean_tokens are two different names right.

Comment (ravi03071991) [reply llama_index/embeddings/huggingface_utils.py:89]: okay. Yeah, the names will be consistent for every model card.

Comment (hatianzhang) [reply llama_index/embeddings/huggingface_utils.py:89]: I mean does the model card for every model from huggingface always use `pooling_mode_cls_token` for cls pooling  and `pooling_mode_mean_tokens ` for mean pooling.
is it possible some models using different naming?

Comment (ravi03071991) [reply llama_index/embeddings/huggingface_utils.py:89]: Nope. Every model uses the same naming convention.

Comment (hatianzhang) [reply llama_index/embeddings/huggingface_utils.py:89]: ic good to know!

**Review by hatianzhang — APPROVED** [review]
lgtm!

**Pull request merged**